apicast/src/policy/cors.lua

```diff
@@ -0,0 +1,37 @@
+local policy = require('policy')
+local _M = policy.new('CORS Policy')
+
+local function set_cors_headers()
+  local origin = ngx.var.http_origin
+
+  if not origin then return end
+
+  ngx.header['Access-Control-Allow-Headers'] = ngx.var.http_access_control_request_headers
```

Comment:
I'm not sure about these 2. Should we return the headers and methods that apicast supports? Or should we just accept everything because we don't know that the upstream api accepts?

Reply:
I think this module should accept a configuration that would whitelist what HTTP methods / Headers are supported. And possibly what origins too and if to pass credentials or not.

But until we have that configuration we can't do much other than just allow all. And even with the configuration there should be a way to configure it as accept all that browser requests.

Reply:
Makes sense. We can use the "configuration" field of a policy to pass this information for now: https://github.com/3scale/apicast/blob/master/apicast/src/configuration.lua#L167

Reply:
Yep. That sounds good! Forgot it is there :)